elasticsearch.md

```diff
@@ -1908,8 +1908,130 @@ indices: ["logstash_@explode{x-indices_csv_string}*", "otherIdx"]
 # -> indices: ["logstash_a*", "logstash_b*", "otherIdx"]
 ```
 
+### Variables functions
+
+A value resolved from a variable may not be valid in some contexts.
+Sometimes, the value from the variable needs some preprocessing before usage.
+For example, a HTTP header value containing the uppercase characters is a wrong candidate for the index name because it has to be a lowercase string.
+We introduced variable functions to overcome these limitations.
+They allow modification of the variable values during the variable resolution ([static](elasticsearch.md#static-variables) and [dynamic](elasticsearch.md#dynamic-variables) variables are supported).
+
+With their help, you can use the HTTP header `x-app: KIBANA` containing uppercase characters in the `indices` rule:
```

Comment:
it's rather an unrealistic example. WDYT about using the `X-Forwarded-User` header as the example instead?